File: applied_crypto copy/playground/lesson1/salt_test.rb
require 'bcrypt'
require 'digest'


def hash256(password, salt, r=1)
    hash = ''
    r.times do 
        hash = Digest::SHA2.new(256).hexdigest "#{hash}#{password}#{salt}"
    end
    hash
end


# salt = BCrypt::Engine.generate_salt
salt = "$2a$12$5Bx/CBG3xoXor78zB8DSRe"
# puts "What is your password?"
# password = gets
password = 'password'

hexdigest = hash256(password, salt, 3)
puts hexdigest
# puts password + salt
x1 = Digest::SHA2.new(256).hexdigest "#{password}#{salt}"
x2 = Digest::SHA2.new(256).hexdigest "#{x1}#{password}#{salt}"
x3 = Digest::SHA2.new(256).hexdigest "#{x2}#{password}#{salt}"

puts "test: #{x3 == hexdigest ? 'passed' : 'failed'}"

File: applied_crypto/playground/lesson1/salt_test.py
import bcrypt
import hashlib


# salt = bcrypt.gensalt()
salt = "$2a$12$5Bx/CBG3xoXor78zB8DSRe"
password = 'password'

def hash(password, salt, r=1):
    to_hash = password + salt
    current_hash = ''
    for _ in range(r):
        current_hash = hashlib.sha256((current_hash + to_hash).encode('utf-8')).hexdigest()
    return current_hash

my_hash = hash(password, salt, 3)
print(my_hash)

print('test')
x1 = hashlib.sha256((password + salt).encode('utf-8')).hexdigest()
x2 = hashlib.sha256((x1 + password + salt).encode('utf-8')).hexdigest()
x3 = hashlib.sha256((x2 + password + salt).encode('utf-8')).hexdigest()
print('passed') if(x3 == my_hash) else print('failed')
